src/main/java/vct/col/veymont/Aldebaran.java

```diff
@@ -0,0 +1,127 @@
+package vct.col.veymont;
+
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.function.BiPredicate;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+public class Aldebaran {
+
+    private final int s0;
+    private final int states;
+    private final Map<Integer, Map<String, List<Integer>>> transitions;
+
+    public Aldebaran(int s0, int states, Map<Integer, Map<String, List<Integer>>> transitions) {
+        this.s0 = s0;
+        this.states = states;
+        this.transitions = transitions;
+    }
+
+    public Aldebaran(Scanner in) {
+        if (!in.hasNextLine()) throw new IllegalArgumentException();
```

Comment:
The files read here are generated in the pass GenerateLTS. I tried to replace this java code with scala code, but failed because my scala variant of the method AldebaranTau.init did not perform, it took way longer. AldebaranTau.init pushes code on a stack to be executed later, to solve that java cannot handle tail-recursion calculations. I attached my scala code; the bad-performing method WellBehavedness.getTauClosure replaces AldebaranTau.init. Any ideas on how to get this working are welcome.

[WellBehavedness.zip](https://github.com/utwente-fmt/vercors/files/6618000/WellBehavedness.zip)

Reply:
I found one bug in my scala code and solved it, but now I get a ```java.lang.OutOfMemoryError: Java heap space```
[WellBehavedness.zip](https://github.com/utwente-fmt/vercors/files/6637120/WellBehavedness.zip)

Reply:
I made an iterative version of the tail-recursive function. The computation time of this iterative variant is comparable to printing the LTS and executing the java code, so I replaced the java code with iterative scala code. I think the code is better maintainable now. I'll leave optimizing the computation time for future work.